### ISSUE TITLE
rpc: enable the RPC middleware API to return "rich" errors 

### ISSUE THREAD

**Issue by Roasbeef** [lightningnetwork/lnd#6729]
Spun off from this comment: https://github.com/lightningnetwork/lnd/pull/6630#discussion_r918441370

With #6630 the RPC interceptor can now intercept error responses (from the server) to modify or even entirely swap out what error the client sees. This approach works for now, but ideally in the future we can allow the interceptor to provide a "nice" error, which is actually an underlying `status.Status` proto struct. 

In order to do this, we'll need to add some additional logic to firsts check if the returned (in memory error) is actually a `Status` struct or not. Based on that, we can then re-serialize it as a normal proto, or just an error string.

**Comment by l0k18** [comment]
This might be another good one for me to take on - I'm very familiar with interfaces and logging systems.

**Comment by KapilSareen** [comment]
hey @guggero are you currently working on this? I'd love to take this up.

**Comment by guggero** [comment]
No, I'm not actively looking into this. You're welcome to take a stab at it.

**Comment by Suvrat1629** [comment]
@guggero Hey since no work is being done on this yet and this is a good first issue is this available to work on?

**Comment by Roasbeef** [comment]
@Suvrat1629 do you have a proposed plan of attack?

**Comment by Suvrat1629** [comment]
@Roasbeef Yes I have come up with a implementation plan it would be great to get it verified by you.
Steps:
1. Add helper to convert common errors to `status.Status` with appropriate gRPC codes (for `ErrTimeoutReached`, `ErrShuttingDown`, `errClientQuit`).

2. In `NewMessageInterceptionRequest`:
    Serialize `status.Status` errors as `google.rpc.Status` proto.
    Set `ProtoTypeName` = "google.rpc.Status".
    Fallback to string for plain errors.
    In sendInterceptRequests:

3. When handling errors, parse as `google.rpc.Status` if `ProtoTypeName` = "google.rpc.Status", else fallback to plain error.

Since I am not very versed with the codebase please verify this approach and tell me if I can proceed with the pr.
Thank you.

**Comment by Suvrat1629** [comment]
@Roasbeef any comments on [this](https://github.com/lightningnetwork/lnd/issues/6729#issuecomment-3289285899) approach?
